autogpt_platform/backend/backend/blocks/compass/triggers.py

```diff
@@ -0,0 +1,75 @@
+from typing import Literal
+from backend.data.block import (
+    Block,
+    BlockCategory,
+    BlockOutput,
+    BlockSchema,
+    BlockWebhookConfig,
+)
+from backend.data.model import CredentialsField, CredentialsMetaInput, SchemaField
+from backend.integrations.webhooks.simple_webhook_manager import CompassWebhookType
+from pydantic import BaseModel
+
+
+class Transcription(BaseModel):
+    text: str
+    speaker: str
+    end: float
+    start: float
+    duration: float
+
+
+class TranscriptionDataModel(BaseModel):
+    date: str
+    transcription: str
+    transcriptions: list[Transcription]
+
+
+class CompassAITriggerBlock(Block):
+    class Input(BlockSchema):
+        class EventsFilter(BaseModel):
+            all: bool = True
+
+        payload: TranscriptionDataModel = SchemaField(hidden=True)
+        events: EventsFilter = SchemaField(
+            description="Filter the events to be triggered on."
+        )
+
+        credentials: CredentialsMetaInput[Literal["compass"], Literal["api_key"]] = (
+            CredentialsField(
+                provider="compass",
+                supported_credential_types={"api_key"},
+            )
+        )
```

Comment:
I don't have any real concept of credentials that I need here. If we generate a secret on the platform side, that would be useful so we can ensure its a trusted source who sent the hook. Maybe we should store that in the credential?

Reply:
For that we already have the [`Webhook.secret`](https://github.com/Significant-Gravitas/AutoGPT/pull/8358/files#diff-8d3c47142a10f464ac75c6e6f1b7c3e2ce76a400f6881b34f11c9088895ac5a1R27). This mechanism doesn't need credentials, so we should just omit the `credentials` field.

Reply:
We'll have to rework this check right here to only enforce that `credentials` is given if there is an actual `credentials` field on `block.input_schema`:

https://github.com/Significant-Gravitas/AutoGPT/blob/68f4aeaf5afdd0ed258e6e84f04b47fffbacf21b/autogpt_platform/backend/backend/integrations/webhooks/graph_lifecycle_hooks.py#L114-L128

This is also one of the two places where the presence of an event filter is enforced.